### PR TITLE
Feature/canvas android

### PR DESCRIPTION
**Pull request by j8seangel**
This PR includes the custom Canvas Android implementation what adds the below features, ONLY IN ANDROID, TODO: IOS
- Offline support,
- Custom canvas filter by dates layer.
- New interaction layer to highlight the area selected (TODO: improve the highlight size in zoom level changes)

This component extends the UrlTile one and receives the following new params:
- areaId: necessary to generate the cached folder.
- alertType: supports VIIRS("viirs") and GLAD ("umd_as_it_happens") layers, also mandatory for the cached folder.
- isConnected: gets the online status in the app to get from the server or from the cached folder
- min and maxDate: dates with the days count from the origin in GLAD('20150101') and 1(today)-8(last-week) to VIIRS.

### CODE REVIEW

**Comment by j8seangel** [comment]
Please, don't merge until the master branch is sync with the airbnb master branch to keep the repos synched and make the conflicts easier to solve.
Also remember to remove `#feature/canvas-android` from the FW repo package.json in `"react-native-maps": "Vizzuality/react-native-maps#feature/canvas-android"`

**Comment by j8seangel** [comment]
Feel free to merge! (https://github.com/Vizzuality/react-native-maps/pull/6)

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasInteractionUrlTileManager.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    .getDefaultDisplay()
+                    .getRealMetrics(metrics);
+        } else {
+            metrics = reactContext.getResources().getDisplayMetrics();
+        }
+    }
+
+    @Override
+    public String getName() {
+        Log.d("Map", "Get name");
```
Comment (sorodrigo) [inline lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasInteractionUrlTileManager.java]: is this a console log?

File: example/examples/CustomTiles.js [hunk 2]
```diff
@@ -36,20 +45,95 @@ class CustomTiles extends React.Component {
...
+
+  updateRegion = (region) => {
+    this.setState({ region });
+  }
+
   render() {
-    const { region } = this.state;
+    const { region, coordinates } = this.state;
+    const hasCoordinates = (coordinates.tile && coordinates.tile.length === 3) || false;
+    console.log(coordinates);
```
Comment (sorodrigo) [inline example/examples/CustomTiles.js]: console log

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasUrlTileManager.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    .getDefaultDisplay()
+                    .getRealMetrics(metrics);
+        } else {
+            metrics = reactContext.getResources().getDisplayMetrics();
+        }
+    }
+
+    @Override
+    public String getName() {
+        Log.d("Map", "Get name");
```
Comment (sorodrigo) [inline lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasUrlTileManager.java]: log

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasTileProvider.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        int srcW = this.width;
+        int srcH = this.height;
+        int scaleSize = 1;
+
+        int minDate = Integer.valueOf(this.minDate);
+        int maxDate = Integer.valueOf(this.maxDate);
+
+        if (this.coordinates != null) {
+            int[] tile = this.coordinates.getTile();
+            Log.d("tile Position: ", x + "-" + y + "-" + zoom);
```
Comment (sorodrigo) [inline lib/android/src/main/java/com/airbnb/android/react/maps/AirMapCanvasTileProvider.java]: log

**Review by sorodrigo — CHANGES_REQUESTED** [review]
Great! I left you some marks on some logs I found.

**Comment by j8seangel** [comment]
Logs removed

**Review by sorodrigo — APPROVED** [review]

**Comment by geriux** [comment]
Awesome work!!!!!!

**Pull request merged**